Feature/187876837 -  Examples indexed db

jeremymccurdy-redspace opened the pull request:
This PR ports over the IndexedDB example from the old springroll.io.

The old example displayed data store contents in a Vue data table modal. I've written a simplified version of that for Docusaurus. It just is a modal with the table, there is no pagination or sorting.

The help section at the bottom of the page has been updated to use Docusaurus styling. The tabs have been put at the top of the section instead of along the side, this was done to keep the layout responsive.

The example iframe has been updated in several ways:
- The styling has been changed to better match the Docusaurus dark mode. It doesn't change when toggling dark mode off, but doesn't clash with light mode.
- The version of Springroll it's using was updated
- The `Delete Store Name` field has been removed, and its functionality is now just covered by the regular `Store Name` field.
- There is a lot more error checking to cover issues with empty fields, and trying to perform DB operations without a connection. 
- Deleting a store  will close active DB connections if there are any before running. This would break things before.

There are still some bugs with opening and deleting stores if you enter the wrong version number. Container doesn't catch or expose these errors, and there's no way to catch them in SR. I'm going to make a bug ticket to cover these.

Ticket: https://www.pivotaltracker.com/story/show/187876837

Screenshots:
![Screenshot 2024-08-22 at 9 55 23 AM](https://github.com/user-attachments/assets/3a021dbe-c333-43c0-b010-76b87b58d59f)
![Screenshot 2024-08-22 at 9 55 42 AM](https://github.com/user-attachments/assets/04fd75ba-d077-474a-8018-2dba9a761943)
![Screenshot 2024-08-22 at 9 58 55 AM](https://github.com/user-attachments/assets/8f6a6a94-39ab-4290-9f22-cd4d3eae3088)

jeremymccurdy-redspace committed:
Merge branch 'feature/187876834-Examples-Controls' into feature/187876837-Examples-IndexedDB

jeremymccurdy-redspace committed:
Ported indexed DB example.

jeremymccurdy-redspace committed:
Adding data table component. It's a very simple replacement for the component in the original example. Infima doesn't have a component for this kind of thing, so I had to make one.

jeremymccurdy-redspace committed:
Data for the example. Split it into its own file for readability and to allow for reuse between components.

jeremymccurdy-redspace committed:
Added a component to represent the docs for the example, a port from the original vue class. Switched it over to use Infima components and to better follow the page styling.

jeremymccurdy-redspace committed:
Updating styling of example, code cleanup & documentation, and lots of error handling/bug fixes. Also updated the version of SpringRoll used by the example iframe.

jeremymccurdy-redspace committed:
Merge branch 'docusaurus-update' into feature/187876837-Examples-IndexedDB

jeremymccurdy-redspace committed:
- Fixed errors with empty fields.
- Formatting fixes, general code cleanup
- Comments
- Fixed SR Container loading

jeremymccurdy-redspace committed:
Fixed spacing and alignment in the help section

jeremymccurdy-redspace committed:
Fixed styling on X button

jeremymccurdy-redspace committed:
Merge branch 'develop' into feature/187876837-Examples-IndexedDB

jeremymccurdy-redspace committed:
Removed commented out style

902seanryan commented:
Love this, thanks for taking the time to fix our implementation. its been just sitting there for years

902seanryan reviewed (approved):
Everything looks killer to me. I do think it might be adding a small section to the readme regarding the UMD files we're using for examples. Because the child pages aren't using them we should probably either add a step to the build that copies them over or, more likely, add a blurb that mentions to avoid future confusion. 

We can do that here or some other PR but I wanna write it down

jeremymccurdy-redspace committed:
Updated readme with UMD info

jeremymccurdy-redspace committed:
Moved IndexedDB SpringRoll UMD file to the same directory as the HTML file to keep things cleaner.

jeremymccurdy-redspace committed:
Added missing .map file for SpringRoll umd

jeremymccurdy-redspace commented:
@902seanryan I've updated the readme. I've also moved the UMD file to the same folder as the IDB HTML file to keep things organized, and added the `.map` file for the UMD. Oh, and I removed one console log that was kind of pointless.
